Sources/StreamChat/ChatClient.swift

```diff
@@ -474,9 +480,14 @@ extension ChatClient {
         var notificationCenterBuilder = EventNotificationCenter.init
         
         var internetConnection: (_ center: NotificationCenter) -> InternetConnection = {
-            InternetConnection(notificationCenter: $0)
+            InternetConnection(notificationCenter: $0, monitor: monitor)
         }
 
+        #if TESTS
+        // Used only in tests
+        static var monitor: InternetConnectionMonitor?
```

Comment:
@evsaev @b-onc this is the place where we lack the feasible dependency injection pattern.

`InternetConnection` provides `monitor` param in its initializer, but currently there's no easy way to specify it.
For now I marked this implementation for tests purposes only. 

Happy to hear your thoughts on this.

Reply:
Discussed and incorporated comments